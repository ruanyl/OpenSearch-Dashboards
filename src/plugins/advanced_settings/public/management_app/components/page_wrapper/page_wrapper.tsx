/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiPageContent } from '@elastic/eui';
import React from 'react';

export const PageWrapper = (props: { children?: React.ReactChild }) => {
  return (
    <EuiPageContent
      style={{ maxWidth: '75%', marginTop: '20px' }}
      hasShadow={false}
      hasBorder={false}
      panelPaddingSize="none"
      horizontalPosition="center"
      color="transparent"
      {...props}
    />
  );
};
