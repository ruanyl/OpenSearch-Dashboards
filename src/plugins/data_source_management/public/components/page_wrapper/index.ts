/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export { PageWrapper } from './page_wrapper';
