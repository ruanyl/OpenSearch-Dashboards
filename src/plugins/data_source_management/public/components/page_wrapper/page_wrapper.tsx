/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiPageContent } from '@elastic/eui';
import React from 'react';

export const PageWrapper = (props: { fullWidth?: boolean; children?: React.ReactChild }) => {
  return (
    <EuiPageContent
      style={props.fullWidth ? {} : { maxWidth: '75%', marginTop: '20px' }}
      hasShadow={false}
      hasBorder={false}
      panelPaddingSize="none"
      horizontalPosition="center"
      color="transparent"
      {...props}
    />
  );
};
