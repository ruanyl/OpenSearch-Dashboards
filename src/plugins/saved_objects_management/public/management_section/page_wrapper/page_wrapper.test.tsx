/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from '@testing-library/react';
import { PageWrapper } from './page_wrapper';

describe('PageWrapper', () => {
  it('should render normally', async () => {
    const { findByText, container } = render(<PageWrapper>Foo</PageWrapper>);
    await findByText('Foo');
    expect(container).toMatchSnapshot();
  });
});
