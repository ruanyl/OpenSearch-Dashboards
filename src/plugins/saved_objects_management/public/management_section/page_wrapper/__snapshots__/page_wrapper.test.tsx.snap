// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageWrapper should render normally 1`] = `
<div>
  <div
    class="euiPanel euiPanel--borderRadiusMedium euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContent euiPageContent--horizontalCenter"
    role="main"
    style="max-width: 75%; margin-top: 20px;"
  >
    Foo
  </div>
</div>
`;
